### PR TITLE
787 fix laser pointer disabling

### LINKED ISSUE #787: When a player on a pure client disables the laser pointer, it still shows on the host and other clients
Run SEE in multi-player mode. Disable the laser pointer (pressing P) on client instance that is not the host. 

Actual result: the laser pointer is disabled locally but not for all other clients.

Expected result: the laser pointer is disabled everywhere.

### PR DESCRIPTION
**Pull request by koschke**
The problem with the laser beamer was actually caused by a wrong implementation in `ActionNetwork`.

This PR corrects this implementation. Now, `ExecuteOnClient` will be called on **every** client, including a client that is a server at the same time, in other words, a host. Such were the original semantics of our first self-made network framework, which was later replaced by Unity's RPC (however, incorrectly in this regard).

When merged, close #787.

### CODE REVIEW

**Review by koschke — COMMENTED** [review]
Ready for review. 

I selected @Cyclone1337 as a reviewer because I made changes to the following two files of which he is the author:

- `Assets/SEE/Net/Actions/Drawable/DrawableNetAction.cs`
- `Assets/SEE/Net/Actions/Drawable/SynchronizeCurrentOrderInLayer.cs`

File: Assets/SEE/Game/Avatars/AvatarAimingSystem.cs [hunk 1]
```diff
@@ -90,14 +90,22 @@ private void TogglePointing()
...
         /// If <paramref name="activate"/> is true, the laser will be turned on;
         /// otherwise turned off.
-        /// This parameter is also propagated to all clients.
+        /// This parameter is also propagated to all clients if not <paramref name="remoteRequest"/>.
         /// </summary>
         /// <param name="activate">whether pointing is to be activated</param>
-        /// <remarks>This method is called either as a interaction request of the local
+        /// <param name="remoteRequest">whether the request comes from a remote
+        /// client, in which case the change of the pointing modus will not be
+        /// propataged to other clients vai <see cref="TogglePointingNetAction"/></param>
```
Comment (Cyclone1337) [inline Assets/SEE/Game/Avatars/AvatarAimingSystem.cs:98]: Typo: vai => via

Comment (Cyclone1337) [reply Assets/SEE/Game/Avatars/AvatarAimingSystem.cs:98]: Typo: propataged => propagated

File: Assets/SEE/Net/ActionNetwork.cs [hunk 2]
```diff
@@ -179,20 +171,15 @@ internal void SyncClientServerRpc(ulong clientId)
         }
 
         /// <summary>
-        /// Executes an action, even if the sender and this client are the same. This is used
+        /// Executes an action on aclient, even if the sender and this client are the same. This is used
```
Comment (Cyclone1337) [inline Assets/SEE/Net/ActionNetwork.cs:174]: missing whitespace

File: Assets/SEE/Net/Actions/AddNodeNetAction.cs [hunk 3]
```diff
@@ -46,19 +46,10 @@ public AddNodeNetAction
              Vector3 scale)
             : base()
```
Comment (Cyclone1337) [inline Assets/SEE/Net/Actions/AddNodeNetAction.cs:47]: In our other review, we already found definitions that also had this kind of formatting of the parameter list, where the opening parenthesis is on a new line. Do you know what causes this formatting?

File: Assets/SEE/Game/Avatars/AvatarAimingSystem.cs [hunk 4]
```diff
@@ -90,14 +90,22 @@ private void TogglePointing()
         /// <summary>
         /// If <paramref name="activate"/> is true, the laser will be turned on;
         /// otherwise turned off.
-        /// This parameter is also propagated to all clients.
+        /// This parameter is also propagated to all clients if not <paramref name="remoteRequest"/>.
         /// </summary>
         /// <param name="activate">whether pointing is to be activated</param>
-        /// <remarks>This method is called either as a interaction request of the local
+        /// <param name="remoteRequest">whether the request comes from a remote
+        /// client, in which case the change of the pointing modus will not be
```
Comment (Cyclone1337) [inline Assets/SEE/Game/Avatars/AvatarAimingSystem.cs:97]: Shouldn't it be `mode`?
[Modus](https://en.wiktionary.org/wiki/modus)

**Review by Cyclone1337 — CHANGES_REQUESTED** [review]
Thank you for your pull request. I have also built and tested the changes and did not encounter any errors. The drawables are synchronizing perfectly as well. (ExecuteOnServer was needed for the synchronization of the drawables.)

File: Assets/SEE/Game/Avatars/AvatarAimingSystem.cs (hunk 1 above)
Comment (koschke) [reply Assets/SEE/Game/Avatars/AvatarAimingSystem.cs:98]: Thanks. Fixed.

File: Assets/SEE/Net/Actions/AddNodeNetAction.cs (hunk 3 above)
Comment (koschke) [reply Assets/SEE/Net/Actions/AddNodeNetAction.cs:47]: Actually, I prefer the opening parenthesis on a new line. A parenthesis  at the end of a line with no other token following looks ugly to me. Well, that may be a matter of taste, but we neither have the opening curly brackets at the end of the line where loops or conditions occur.

File: Assets/SEE/Game/Avatars/AvatarAimingSystem.cs (hunk 4 above)
Comment (koschke) [reply Assets/SEE/Game/Avatars/AvatarAimingSystem.cs:97]: Yes, mode would be better. Fixed.

**Comment by koschke** [comment]
Closes #787

**Pull request closed without merging**